mlir/lib/Tools/PDLL/Parser/Parser.cpp

```diff
@@ -2559,9 +2760,12 @@ FailureOr<ast::Stmt *> Parser::parseStmt(bool expectTerminalSemicolon) {
   case Token::kw_erase:
     stmt = parseEraseStmt();
     break;
-  case Token::kw_let:
+  case Token::kw_let: {
     stmt = parseLetStmt();
+    llvm::SaveAndRestore saveCtx(nativeOperatorContext,
```

Comment:
Isn't `SaveAndRestore` doing RAII? I wonder if we still need to set it back to `Generic`, since the value should be restored once it leaves `parseLetStmt`.